File: my-angular-app/src/app/@shared/models/instruction.ts
import { product } from "./Product"
import { GetInstructions } from "./instructions"

export class form {

  constructor(data: GetInstructions) {
    this.client = data.clientName
    this.clientRef = data.clientRef
    this.billingRef = data.billingRef
    this.status = data.status
    this.pickUpAddress = data.pickupAddress
    this.deliveryAddress = data.deliveryAddress
  }
  client: string = "";
  clientRef: string = "";
  billingRef: string = "";
  status: string = "";
  pickUpAddress: string = "";
  deliveryAddress: string = "";
  product: string = "";
  qty: number = 0;
  productCode: string = "";
  productDiscription: string = "";
}


export class PostInstruction {

  constructor(formdata: form, productlist: product[]) {
    this.clientName = formdata.client
    this.pickupAddress = formdata.pickUpAddress
    this.deliveryAddress = formdata.deliveryAddress
    this.clientRef = formdata.clientRef
    this.billingRef = formdata.billingRef
    this.products = productlist
  }
  clientName: string = "";
  pickupAddress: string = "";
  deliveryAddress: string = "";
  clientRef: string = "";
  billingRef: string = "";
  products: product[] = [];
}

File: ScheduleTransportationInstructions/Domain/Entities/Instruction.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Instruction
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public DateTime InstructionDate { get; set; } = DateTime.Now;
        //Auto populate but allow for changes

        [Required]
        public string ClientName { get; set; }
        //Select from a client list

        [Required]
        public string PickupAddress { get; set; }

        [Required]
        public string DeliveryAddress { get; set; }

        //Client reference
        [Required]
        public string ClientRef { get; set; }

        [Required]
        public string BillingRef { get; set; }
        //Billing references are used to obtain charge rates

        [Required]
        public string Status { get; set; }
        //When instruction is captured, set status to “Pending”
       

        public IEnumerable<Product> Products { get; set; }
        public IEnumerable<ScheduleTransport> Transporter { get; set; }
    }
}
